internal/dag/gatewayapi_processor.go

```diff
@@ -149,10 +150,12 @@ func selectorMatches(selector metav1.LabelSelector, objLabels map[string]string)
 }
 
 func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRoute) {
+	routeAccessor, commit := p.dag.StatusCache.HTTPRouteAccessor(route)
+	defer commit()
 
 	// Validate TLS Configuration
 	if route.Spec.TLS != nil {
-		p.Error("NOT IMPLEMENTED: The 'RouteTLSConfig' is not yet implemented.")
+		routeAccessor.ConditionFor(status.NotImplemented, "NotImplemented", "HTTPRoute.Spec.TLS: Not yet implemented.")
```

Comment:
Maybe worth calling out somewhere that we are adding additional conditions with details about why a Route is invalid, I forget if the original design doc mentioned this specifically? Or if it was assumed the reason/description on the "Admitted" reason would have all this info?

Reply:
Another point for not using status as message passing is that it could be relatively unwieldy it seems, maybe keying off of reasons is good enough, but if you need more detail than that you would want some structured description (JSON probably?)

Reply:
Echoing @sunjayBhatia's q on the conditions, "Admitted" is the only type that Gateway API documents, so should we only use that? I'm not sure, I guess it depends on who is consuming the info.

Reply:
So, it's okay for implementations to add their own Conditions, but they should roll up into the `Admitted` condition. I think that Steve has handled that here in another way, but I think it would be best to be clear about what we're doing, and make sure that the status gets rolled up into the `Admitted` condition at the end (we can use the `commit` function to do that).

I think that `ConditionFor` is not a good name for this one, as it's really doing an `AddCondition` to the proxy update.

Reply:
If we only use `Admitted` then we can only set a single status on the resource? I was under the impression that since it's a set, we'd want to list out all the issues with a route and not just a single.